cognee/tasks/web_scraper/bs4_crawler.py

```
@@ -0,0 +1,546 @@
+
+    async def __aenter__(self):
+        """Enter the context manager, initializing the HTTP client."""
+        await self._ensure_client()
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        """Exit the context manager, closing the HTTP client."""
+        await self.close()
+
+    @lru_cache(maxsize=1024)
+    def _domain_from_url(url: str) -> str:
```

Pylint (3.3.8)

[error] Method '_domain_from_url' should have "self" as first argument

(E0213)